File: library/src/main/java/me/proxer/library/api/ucp/WatchedEpisodesEndpoint.java
package me.proxer.library.api.ucp;

import lombok.experimental.Accessors;
import me.proxer.library.api.Endpoint;
import me.proxer.library.api.ProxerCall;

/**
 * Endpoint for requesting the amount of watched episodes of the current user.
 *
 * @author Ruben Gees
 */
@Accessors(fluent = true)
public final class WatchedEpisodesEndpoint implements Endpoint<Integer> {

    private final InternalApi internalApi;

    WatchedEpisodesEndpoint(final InternalApi internalApi) {
        this.internalApi = internalApi;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public ProxerCall<Integer> build() {
        return internalApi.watchedEpisodes();
    }
}

File: library/src/main/kotlin/me/proxer/library/api/ucp/WatchedEpisodesEndpoint.kt
package me.proxer.library.api.ucp

import me.proxer.library.ProxerCall
import me.proxer.library.api.Endpoint

/**
 * Endpoint for requesting the amount of watched episodes of the current user.
 *
 * @author Ruben Gees
 */
class WatchedEpisodesEndpoint internal constructor(
    private val internalApi: InternalApi
) : Endpoint<Int> {

    override fun build(): ProxerCall<Int> {
        return internalApi.watchedEpisodes()
    }
}
